app/host/system_tasks.py

```diff
@@ -40,56 +41,56 @@ def run_process(self, interval_minutes=200):
-                if not saved_transient.redshift and transient.redshift:
-                    tasks = TaskRegister.objects.filter(
-                        transient=saved_transient)
-                    for t in tasks:
-                        ### TODO: this could be smarter
-                        ### we don't *always* need to re-process every stage
-                        t.status = Status.objects.get(message="not processed")
-                        t.save()
-                # update info
-                new_transient_dict = transient.__dict__
-                if "host_id" in new_transient_dict.keys():
-                    if saved_transient.host_id is not None:
-                        new_transient_dict["host_id"] = saved_transient.host_id
-                    else:
-                        del new_transient_dict["host_id"]
-
-                keys_to_del = [
-                    "_state",
-                    "id",
-                    "progress",
-                    "tasks_initialized",
-                    "photometric_class",
-                    "milkyway_dust_reddening",
-                    "processing_status",
-                ]
-                for k in keys_to_del:
-                    del new_transient_dict[k]
-                new_transient_dict['tasks_initialized'] = False
-                saved_transients.filter(name__exact=transient.name).update(
-                    **new_transient_dict
-                )
-
-            except Transient.DoesNotExist:
+            # If the transient has not already been ingested, save the TNS
+            # data and proceed to the next transient
+            saved_transient = saved_transients.filter(name__exact=transient.name)
+            if not saved_transient:
                 transient.save()
                 count += 1
+                continue
+            # If the transient was previously ingested, compare to the current TNS data.
+            saved_transient = saved_transient[0]
+
+            # If the timestamps are identical, assume no further processing is necessary.
+            if (saved_transient.public_timestamp.replace(tzinfo=None) - parser.parse(
+                transient.public_timestamp
+            ) == datetime.timedelta(0)):
+                continue
+
+            # Update the saved transient data with the latest TNS data after some processing.
+            new_transient_dict = transient.__dict__
+            # If the incoming data includes a "host_id", then the "transient" object being ingested is not
+            # actually from TNS but is instead from an internal source from a special execution of this
+            # function. In this case, if a host has already been associate with the saved transient, update
+            # the value to the new one; however, if the saved transient does not have a host match, then
+            # discard the new value.
+            if "host_id" in new_transient_dict.keys():
```

Comment:
@djones1040 What is going on here? The comment I added is my best guess, given that there is no way an actual TNS API response would include a Django-model-specific foreign key ID linking the Transient and Host database tables.

Reply:
After discussion we agreed to eliminate this cruft in 9a3d8ac